### DISCUSSION TITLE
Getting false values in entity views for attributes which are actually true when using multiselect

### DISCUSSION THREAD

**Discussion by bujikun** [Blazebit/blaze-persistence#1863] [General]
Hello, guys! I have come across this problem whenever i use multiselect in entity views i end up with false values always even for columns that i actually know hold true values. 

I have sample entities below

```java
@Entity
@Table(name = "posts")
public class Post  {
    private String title;
    private String body;
    private boolean active;

    @OneToMany(mappedBy = "post",fetch = FetchType.LAZY,cascade = {CascadeType.MERGE,CascadeType.PERSIST})
    private List<Comment> comments = new ArrayList<>();

    public void addComment(Comment comment){
        this.comments.add(comment);
        comment.setPost(this);
    }
}

@Entity
@Table(name = "comments")
public class Comment{
    private String content;
    private boolean active;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id",referencedColumnName = "id")
    private Post post;

}
```

And here is my entity view to get all comments in a post using multiselect

```java
@EntityView(Post.class)
public interface PostView {
    @IdMapping
    Long getId();
    String getTitle();
    String getBody();
    boolean isActive();
    @Mapping(fetch = FetchStrategy.MULTISET)
    Set<CommentView> getComments();

    @EntityView(Comment.class)
     interface CommentView {
        @IdMapping
        Long getId();
        String getContent();
        boolean isActive();

    }

}
```

But this returns all comments with active false even for the ones it is true. The active column in PostView is not affected by this.
This also does not happen when i use FetchStrategy.JOIN.

I tried to investigate the query generated and i am still wondering if this is caused by casting to char as opposed to casting to int ( not sure how the results are transformed when using a multiset)

```sql
select
   p1_0.id,
   p1_0.title,
   p1_0.body,
   p1_0.active,
    (select
         json_arrayagg(json_object('f0', cast(c1_0.id as char), 'f1', cast(c1_0.active as char), 'f2', cast(c1_0.content as char)))
     from
         comments c1_0
     where
        p1_0.id=c1_0.fk_post_id)
from
    posts p1_0;
```

Does anyone have an idea why this is happening?

**Comment by beikov** [comment]
Hey, can you set a break point into `com.blazebit.persistence.view.impl.type.BooleanBasicUserType#fromString` please? This is where the string that the `cast` produces is supposed to be transformed to a `boolean`.
Are you using MySQL? If so, the problem might be that MySQL has no native `boolean` type and the cast expression produces `1`/`0` instead of the expected `t`/`f`. If that's the case, then the fix might be as simple as changing `com.blazebit.persistence.view.impl.type.BooleanBasicUserType#fromString` to also accept `'1'` as valid value for `true`.
